src/core/id-headers.js

```diff
@@ -20,8 +28,15 @@ export function run(conf) {
       id = h.parentElement.id || h.id;
     }
     if (!conf.addSectionLinks) continue;
-    h.appendChild(html`
-      <a href="${`#${id}`}" class="self-link" aria-label="§"></a>
+    let label = h.closest(".appendix")
+      ? l10n.perma_for_appendix
+      : l10n.perma_for_section;
+    const sectionNumber = h.querySelector(":scope > bdi.secno");
+    if (sectionNumber) {
+      label += ` ${norm(sectionNumber.textContent)}`;
```

Comment:
Nit: maybe we should we add the sectionNumber as a data attribute on `<bdi>` in `core/structure`? Could be nice as we've also similar thing in `core/dfn-index`.

Reply:
Yeah, it's same same.
https://github.com/w3c/respec/blob/34242b5f079ba209433966aebb9c6255d35a9d0e/src/core/dfn-index.js#L225-L227

Reply:
Though, feel free to ignore this suggestion. Non-blocking.